rest-service/manager_rest/security/secured_resource.py

```diff
@@ -18,14 +18,15 @@
 from flask_restful import Resource
 from flask import request, current_app, Response, jsonify
 
-from manager_rest.utils import abort_error
+from manager_rest.utils import abort_error, traces
 from manager_rest.manager_exceptions import MissingPremiumPackage
 
 from .authentication import authenticator
 
 
 def authenticate(func):
     @wraps(func)
+    @traces('authenticate')
```

Comment:
This should work if your wrap 'authenticate' func instead?

Reply:
Then the decoration is performed out of context and therefore it won't have access to the `tracer` obj

Reply:
We can trace two things here, either when the decorator was called (it'll be out of flask app context) _or_ trace when the inner function runs (I think we should be tracing the inner function).
To trace the inner function execution - the function that executes every time the wrapped function is executed, I'd have to place the @traces decorator where it's at.